src/core/source_buffers/source_buffers_store.ts

```diff
@@ -114,25 +140,93 @@ export default class SourceBuffersStore {
+   *
+   * This function will return  an object containing the key `type` which can
+   * be equal to either one of those three value:
+   *
+   *   - "set": A SourceBuffer has been created. You will in this case also have
+   *     a second key, `value`, which will contain the related
+   *     QueuedSourceBuffer instance.
+   *     Please note that you will need to wait until `this.onSourceBuffersReady()`
+   *     has emitted before updating a native SourceBuffer.
+   *
+   *   - "disabled": The SourceBuffer has been explicitely disabled for this
+   *     type.
+   *
+   *   - "unset": No action has been taken yet for that SourceBuffer.
    *
    * @param {string} bufferType
    * @returns {QueuedSourceBuffer|null}
    */
-  public get(bufferType : IBufferType) : QueuedSourceBuffer<any>|null {
+  public getStatus(bufferType : IBufferType) : { type : "set";
+                                                 value : QueuedSourceBuffer<any>; } |
+                                               { type : "unset" } |
+                                               { type : "disabled" }
+  {
     const initializedBuffer = this._initializedSourceBuffers[bufferType];
-    return initializedBuffer != null ? initializedBuffer :
-                                       null;
+    return initializedBuffer === undefined ? { type: "unset" } :
+           initializedBuffer === null      ? { type: "disabled" } :
+                                             { type: "set",
+                                               value: initializedBuffer };
+  }
+
+  /**
+   * Native SourceBuffers (audio and video) need to all be created before they
+   * can be used.
+   *
+   * From https://w3c.github.io/media-source/#methods
+   *   For example, a user agent may throw a QuotaExceededError
+   *   exception if the media element has reached the HAVE_METADATA
+   *   readyState. This can occur if the user agent's media engine
+   *   does not support adding more tracks during playback.
+   *
+   * This function will return an Observable emitting when each of these
+   * SourceBuffers is either created or disabled.
+   * @return {Observable}
+   */
+  public onSourceBuffersReady() : Observable<void> {
+    if (this._areNativeSourceBuffersReady()) {
+      return observableOf(undefined);
+    }
+    return new Observable(obs => {
+      this._onNativeSourceBufferAddedOrDisabled.push(() => {
+        if (this._areNativeSourceBuffersReady()) {
+          obs.next(undefined);
+          obs.complete();
+        }
+      });
+    });
+  }
+
+  /**
+   * Explicitely set no SourceBuffer for a given buffer type.
+   * A call to this function is needed at least for unused native buffer types
+   * ("audio" and "video"), to be able to emit through `onSourceBuffersReady`
+   * when both of those are set.
+   * @param {string}
+   */
+  public disableSourceBuffer(bufferType : IBufferType) : void {
+    if (this._initializedSourceBuffers[bufferType] !== undefined) {
```

Comment:
So if it is already disable, you throw ?

Reply:
Yeah, I hesitated. I thought that disabling an already disabled SourceBuffer would always be due to a problem but you may be right that we have no reason to throw in that case.

What do you think we should do?

Reply:
In my opinion, even if we want to disable inadvertently an already disabled SourceBuffer, the function should not throw. In these cases, the problem is elsewhere and the source buffer store should just do nothing, as it is not supposed to know that there is a problem. At least it could warn about it, but not throw.

Reply:
Ok, I agree, done. I now `log.warn` and return in that case.